h/search/logging_filters.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+"""Logging Filters."""
+
+import logging
+
+
+class FilterExceptions(logging.Filter):
+    """Filter out the specified exceptions with specified logging level."""
+
+    def __init__(self, ignore_exceptions):
+        """
+        Configure filtering out of the specified exceptions with specified logging level.
+
+        ignore_exceptions: a tuple of tuples ((exception type, loglevel))
+                           example: (("requests.exceptions.ReadTimeout", "WARNING"),)
```

Comment:
What was the rationale for making `ignore_exceptions` a list of string tuples rather than say, a list of `(exception class, numeric logging level)`, eg. `[(requests.exceptions.ReadTimeout, logging.WARNING)]`?

There are several lines of code here and test cases needed to check the (string, string) => (type, number) conversion and you might be able to avoid that by taking the type + number directly.

Reply:
I did this thinking that this is purely configuration and there does exist a python dict configuration for loggers. If we ever switch to that kind of config then I don't think we can use a raw import statement. Note this does except both string and number type for the log level. I can add a specific test indicating that as I don't think I have one right now.

Reply:
> I did this thinking that this is purely configuration and there does exist a python dict configuration for loggers.

Ah, OK, I see. My bias would be to apply the YAGNI principle here, but I'll leave this up to you to make a call.